test/jasmine/tests/plot_interact_test.js

```diff
@@ -0,0 +1,177 @@
+var d3 = require('d3');
+
+var Plotly = require('@src/index');
+var Fx = require('@src/plots/cartesian/graph_interact');
+
+
+describe('Test plot structure', function () {
+    'use strict';
+
+    function createGraphDiv() {
+        var gd = document.createElement('div');
+        gd.id = 'graph';
+        document.body.appendChild(gd);
+        return gd;
+    }
+
+    function destroyGraphDiv() {
+        var gd = document.getElementById('graph');
+        document.body.removeChild(gd);
+    }
+
+    afterEach(destroyGraphDiv);
+
+    describe('cartesian plots', function() {
+        describe('scatter traces', function() {
+            var mock = require('@mocks/14.json');
+
+            beforeEach(function(done) {
+                Plotly.plot(createGraphDiv(), mock.data, mock.layout).then(done);
+            });
+
+            it('has one *subplot xy* node', function() {
+                var nodes = d3.selectAll('g.subplot.xy');
+                expect(nodes[0].length).toEqual(1);
+            });
+
+            it('has one *scatterlayer* node', function() {
+                var nodes = d3.selectAll('g.scatterlayer');
+                expect(nodes[0].length).toEqual(1);
+            });
+
+            it('has as many *trace scatter* nodes as there are traces', function() {
+                var nodes = d3.selectAll('g.trace.scatter');
+                expect(nodes[0].length).toEqual(mock.data.length);
+            });
+
+            it('has as many *point* nodes as there are traces', function() {
+                var nodes = d3.selectAll('path.point');
+
+                var Npts = 0;
+                mock.data.forEach(function(trace) {
+                    Npts += trace.x.length;
+                });
+
+                expect(nodes[0].length).toEqual(Npts);
+            });
+
+            it('responds to hover', function() {
+                var gd = document.getElementById('graph');
+
+                var evt = {
+                    clientX: gd.layout.width/ 2,
+                    clientY: gd.layout.height / 2
+                };
+
+                Fx.hover('graph', evt, 'xy');
+
+                var hoverTrace = gd._hoverdata[0];
+
+                expect(hoverTrace.curveNumber).toEqual(0);
+                expect(hoverTrace.pointNumber).toEqual(17);
+                expect(hoverTrace.x).toEqual(0.388);
+                expect(hoverTrace.y).toEqual(1);
+
+                expect(d3.selectAll('g.axistext')[0].length).toEqual(1);
```

Comment:
Proof of concept:

![image](https://cloud.githubusercontent.com/assets/6675409/11536802/123fe3f8-98e8-11e5-9f6a-8a406dc557f3.png)